elastic/apm-server
# Ilm alias

## Motivation/summary
With this PR we allow users to customize the ILM `rollover_alias` for every event type. 
It also improves the ILM specific config handling and separates code for unmanaged indices into a dedicated package for easier distinction between setup for ILM and unmanaged (old) indices. 

### Docs Update
Users need to be aware that whenever they switch to a different prefix than `apm-version*` they also need to modify `setup.template.name` and `setup.template.pattern` as otherwise the generic index template will not match the event indices. This is similar to what's already true when users configure `output.elasticsearch.indices`. 
@bmorelli25 I'd appreciate your eyes on the docs changes I made.

## Checklist

-~[ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).~

I have considered changes for:
-~[ ] logging (add log lines, choose appropriate log selector, etc.)~
-~[ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
-~[ ] telemetry~
- [ ] Elasticsearch Service (https://cloud.elastic.co): docs need to be updated (will be done in a follow up when this is merged)
-~[ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
-~[ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes
Configure different settings for `apm-server.ilm.setup.mapping` and check that event aliases, policies, indices and templates get set up accordingly. The event specific templates must match against the customized indices and reference the aliases correctly. 

## Related issues
#3779
